indico/modules/events/layout/controllers/layout.py

```diff
@@ -65,12 +69,50 @@ def _process_lecture_meeting(self):
             return redirect(url_for('.index', self.event_new))
         return WPLayoutEdit.render_template('layout_meeting_lecture.html', self._conf, form=form, event=self.event_new)
 
+    def _make_weeks_view_form(self):
+        form_class = type(b'WeeksViewForm', (IndicoForm,), {})
+        fields = theme_settings.themes['indico-weeks-view']['arguments']
```

Comment:
Since we only keep settings for the currently *selected* theme (ie saved for the event, not for a theme specified with `?view=...`) you can just read it from the settings proxy: `layout_settings.get(event, 'timetable_theme')`